tmail-backend/rate-limiter/rate-limiter-api/src/main/scala/com/linagora/tmail/rate/limiter/api/CacheRateLimitingPlan.scala

```diff
@@ -0,0 +1,52 @@
+package com.linagora.tmail.rate.limiter.api
+
+import com.github.benmanes.caffeine.cache.{Cache, Caffeine}
+import com.linagora.tmail.rate.limiter.api.CacheRateLimitingPlan.{CACHE_EXPIRE_DURATION, LIST_KEY}
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.lang
+import java.time.Duration
+
+object CacheRateLimitingPlan {
+  val CACHE_EXPIRE_DURATION: Duration = Duration.ofHours(2)
+  val LIST_KEY: String = "ALL"
+}
+
+class CacheRateLimitingPlan(repository: RateLimitingPlanRepository) extends RateLimitingPlanRepository {
+
+  val cacheGet: Cache[RateLimitingPlanId, RateLimitingPlan] = Caffeine.newBuilder()
+    .expireAfterWrite(CACHE_EXPIRE_DURATION)
+    .build((planId: RateLimitingPlanId) => SMono.fromPublisher(repository.get(planId)).block())
+
+  val cachePlanExists: Cache[RateLimitingPlanId, java.lang.Boolean] = Caffeine.newBuilder()
+    .expireAfterWrite(CACHE_EXPIRE_DURATION)
+    .build((planId: RateLimitingPlanId) => SMono.fromPublisher(repository.planExists(planId)).block())
+
+  val cacheList: Cache[String, Seq[RateLimitingPlan]] = Caffeine.newBuilder()
+    .expireAfterWrite(CACHE_EXPIRE_DURATION)
+    .build((_: String) => SFlux.fromPublisher(repository.list()).collectSeq().block())
+
+  override def create(creationRequest: RateLimitingPlanCreateRequest): Publisher[RateLimitingPlan] =
+    SMono.fromPublisher(repository.create(creationRequest))
+      .doOnNext(_ => invalidCache())
+
+  override def update(resetRequest: RateLimitingPlanResetRequest): Publisher[Unit] =
+    SMono.fromPublisher(repository.update(resetRequest))
+      .`then`(SMono.fromCallable(() => invalidCache()))
+
+  override def get(id: RateLimitingPlanId): Publisher[RateLimitingPlan] =
+    SMono.justOrEmpty(cacheGet.get(id, planId => SMono.fromPublisher(repository.get(planId)).block()))
+
+  override def planExists(id: RateLimitingPlanId): Publisher[lang.Boolean] =
+    SMono.justOrEmpty(cachePlanExists.get(id, planId => SMono.fromPublisher(repository.planExists(planId)).block()))
+
+  override def list(): Publisher[RateLimitingPlan] =
+    SFlux.fromIterable(cacheList.get(LIST_KEY, _ => SFlux.fromPublisher(repository.list()).collectSeq().block()))
+
+  private def invalidCache(): Unit = {
```

Comment:
```suggestion
  private def invalidateCache(): Unit = {
```

Cool this works for the local node but how about remote nodes? 

(solution: lower caching duration to a few minutes max)